src/main/kotlin/io/github/yuutoproject/yuutobot/commands/Minigame.kt

```diff
@@ -0,0 +1,169 @@
+import io.github.yuutoproject.yuutobot.commands.base.AbstractCommand
+import io.github.yuutoproject.yuutobot.commands.base.CommandCategory
+import net.dv8tion.jda.api.EmbedBuilder
+import net.dv8tion.jda.api.JDA
+import net.dv8tion.jda.api.entities.Emote
+import net.dv8tion.jda.api.entities.Message
+import net.dv8tion.jda.api.entities.MessageEmbed
+import net.dv8tion.jda.api.entities.User
+import net.dv8tion.jda.api.events.message.guild.GuildMessageReceivedEvent
+import net.dv8tion.jda.api.events.message.guild.react.GuildMessageReactionAddEvent
+import net.dv8tion.jda.api.events.message.guild.react.GuildMessageReactionRemoveEvent
+import net.dv8tion.jda.api.hooks.ListenerAdapter
+import java.awt.Color
+
+enum class State {
+    OFF,
+    STARTING,
+    IN_PROGRESS
+}
+
+class MinigameListener(val minigame: Minigame) : ListenerAdapter() {
+    override fun onGuildMessageReceived(event: GuildMessageReceivedEvent) {
+        minigame.messageRecv(event)
+    }
+
+    override fun onGuildMessageReactionAdd(event: GuildMessageReactionAddEvent) {
+        minigame.reactionRecv(event)
+    }
+
+    override fun onGuildMessageReactionRemove(event: GuildMessageReactionRemoveEvent) {
+        minigame.reactionRetr(event)
+    }
+}
+
+class Minigame : AbstractCommand(
+    "minigame",
+    CommandCategory.GAME,
+    "Play a fun quiz with your friends!",
+    "Run `minigame` to begin a new game, and react within the countdown to join.\nRun `minigame skip` to skip a question you do not wish to answer."
+) {
+    var state = State.OFF
+    var players = mutableMapOf<User, Int>()
+    var countdown = 0
+    lateinit var startingMessage: Message
+    var questions = null
+    var currentQuestion = null
+    var answers = null
+    var rounds = 0
+    var timer = System.currentTimeMillis()
+    
+    val messageListener = MinigameListener(this)
+    
+    fun initialize() {
+        state = State.OFF
+        players = mutableMapOf<User, Int>()
+        countdown = 0
+        questions = null
+        currentQuestion = null
+        answers = null
+        rounds = 0
+        timer = System.currentTimeMillis()
+    }
+    
+    override fun run(args: MutableList<String>, event: GuildMessageReceivedEvent) {
+        if (
+            state == State.IN_PROGRESS &&
+                args[0] == "skip" &&
+                players.contains(event.author)
+            ) {
+            event.channel.sendMessage("Skipping the question...").queue()
+//            progress()
+            return
+        }
+
+        if (
+            state == State.IN_PROGRESS &&
+            System.currentTimeMillis() - timer > 3000
+            ) {
+            event.channel.sendMessage("Cancelling stale game...")
+            clean(event.jda)
+            return
+        }
+
+        if (state != State.OFF) {
+            event.channel.sendMessage("A game is already running!")
+            return
+        }
+
+        this.initialize()
+
+        event.jda.addEventListener(messageListener)
```

Comment:
Don't do this, it's bad practice to keep adding and removing listeners, yuuto NEEDS to work in more than one server at the same time